xs/src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -54,7 +71,7 @@ PerimeterGenerator::process()
     for (Surfaces::const_iterator surface = this->slices->surfaces.begin();
         surface != this->slices->surfaces.end(); ++surface) {
         // detect how many perimeters must be generated for this island
-        const int loop_number = this->config->perimeters + surface->extra_perimeters -1;  // 0-indexed loops
+        const int loop_number = num_loops(this->config->perimeters, surface->extra_perimeters);//this->config->perimeters + surface->extra_perimeters-1;  // 0-indexed loops
```

Comment:
By embedding it in the main method you can also cache `min_shell_thickness` and `this->ext_perimeter_flow.scaled_width()` without recalculating them every time.